scripts/prerelease/generate_prerelease_script.py

```diff
@@ -77,6 +78,12 @@ def main(argv=sys.argv[1:]):
     build_files = get_source_build_files(config, args.rosdistro_name)
     build_file = build_files[args.source_build_name]
 
+    # Use targets defined by corresponding Release config instead of
+    # targets defined by Source config. This is to increase the number of
+    # supported targets. New targets are explicitly passed to configure_devel_job().
+    release_build_files = get_release_build_files(config, args.rosdistro_name)
+    release_build_file = release_build_files[args.source_build_name]
```

Comment:
The problem is that there is no correlation between the `build_name` you are using for `source_builds` and `release_builds`. The following `index.yaml` is perfectly valid:

    distributions:
      jade:
        release_builds:
          some_release_build: jade/some-release-build.yaml
          other_release_build: jade/other-release-build.yaml
        source_builds:
          default_source_builds: jade/default-source-build.yaml

Without considering the targets of all release build files you won't be able to use more then the small subset of targets found in the source build file.